### DISCUSSION TITLE
Issue rendering results from Solr

### DISCUSSION THREAD

**Discussion by charlesneedham** [searxng/searxng#2365] [General]
I have a problem. When I search a solr store using searxng, everything works really well except the format of the results. I know I must be missing something fundemental here but I am struggling with this. I have removed some of the info below with asterix's

TITLE: Testing fax functionality
URL: http://knowledgebase-healthcare-in.*****
CONTENT:  Testing fax functionality A <span class="highlight">test</span> script is available for verifying that ******
OPEN_GROUP: True

As you can see it would be great if I was able to have the URL as a link instead of plain text. However the bigger problem here is that the results in the content key value pair have hightlighted the matching keywords. However the web page has not rendered these tags and just displays them as plain text. When I look into the source I can see it has escaped the <> characters in the tag so the source html in the searxng web browser is actually this.

 &lt;span class="highlight"&gt;test&lt;/span&gt; 

Checking solr itself I can see that the content field does not have this hightlighting it is plain text with only \n format characters. So I guess this is being added by Searxng. 

Could you give me some advice/point me in the right direction?

**Comment by return42** [comment]
> well except the format of the results

Not sure what you expect / I don't have a Solr instance at hand to test .. for the discussion; could you post screenshots of what you got from SearXNG (and what you expected).

**Comment by kent57** [comment]
While I am certainly no expert, I am running Apache SOLR on my Centos 7 Linux server and quiring that data using SearXNG.  In order to get the SOLR results to appear properly in the SearXNG search results I needed o add code to the solr.py file.  Essentially, you need to add URL, title and content (plus publishedDate).  In addition, I specified the r['template'] = 'default.html' .   I can help you with this if you need additional information.  Temporarily, you can goto conservativesearch.info and the default query engine is my SOLR engine.  Enter something like 'today' as the search and you will see my solr results.

**Reply by return42** [reply]
@kent57 can you post your solutions .. may we can improve solr.py or add some documentation.

> I specified the `r['template'] = 'default.html'`
> Essentially, you need to add URL, title and content (plus publishedDate). 

By default, the solr engine uses the `key-value.html` template ..
https://github.com/searxng/searxng/blob/7c80807bb8204e9ef0df9a39207e30587c856c0a/searx/engines/solr.py#L84

Sadly we have an issue with this template:

- https://github.com/searxng/searxng/issues/3130

I think we have to fix the issue first, however ..

>  I can help you with this if you need additional information. 

I would like to see your solution .. may we can improve the engine based on your solution.

**Reply by kent57** [reply]
Yes, the default template was key-value.html.   Instead I changed it to 'default.html'.
[solr.py.txt](https://github.com/searxng/searxng/files/14057176/solr.py.txt)


/etc/searxng/settings.yml 

  - name: csearch
    engine: solr
    shortcut: cs
    base_url: 'http://conservativesearch.info:8983'
    weight : 9.0
    defType: 'dismax'
    mm: '3<75%'
    q.op: 'OR'
    collection: 'csearch'
    sort: datepublished desc,score desc
    field_list: website,article_link,article_h2,article_snippet,article_date,website_key,favicon,score
    default_fields: 'combined'
    query_fields: ''
    paging: true
    enable_http: true
    disabled: false
    rows: 20
    time_range_support: true
    categories: [conservative]

Temporary website reference:   https://conservativesearch.info


*** I am more than happy to contribute any way I can.......
